src/core/types.ts

```diff
@@ -48,6 +48,12 @@ export interface NodeInfo {
   };
 }
 
+export interface Node {
+  // id: string; - can be added later once it's useful as input to other methods
+  // pkg: PkgInfo; - it can be nice to return this, consider when exposing more node related methods
+  info?: NodeInfo;
```

Comment:
why is `info` optional?

Reply:
good question. maybe indeed nicer to always have it to not force the user do double nullish-checks. I did it optional just as it's optional on the existing places (`GraphNode` & the `DepGraphData`)

Reply:
Cool. BTW, I'm not a big fan of starting to expose graph/node stuff just for the sake of potential future uses. I'd be happier with `getPkgNodes` being `getPkgNodeInfos`, and only exposing users to the `info` objects. But I guess you've considered this already and see value in starting to leak the graph internals

Reply:
yes ... eventually to allow consumers to traverse the graph, exposing the nodes is necessary I think. Maybe it can be done without exposing the actual ids, but this PR still doesn't expose these.
I do want to open a draft PR with a more complete interface which also allows graph traversal for discussing the bigger design there

Reply:
@darscan the draft PR promised in ☝️ : https://github.com/snyk/dep-graph/pull/47